packages/@sfx/dom-events-plugin/src/dom-events-plugin.ts

```diff
@@ -0,0 +1,140 @@
+   * both the default options and any options passed in. Also the constructor
+   * will accept a window object and a CustomEvent constructor provided
+   * through the options object.
+   *
+   * If either the window object or CustomEvent constructor provided are
+   * not valid, the plugin constructor will throw an error.
+   *
+   * @param options an object that must contain a window property that is
+   * a reference to the browser Window object.
+   */
+  constructor(options: Partial<DomEventsPluginOptions> = {}) {
+    this.options = {...this.options, ...options};
+    this.window = this.options.window;
+    this.CustomEvent = this.options.CustomEvent;
+
+    if (!this.window) {
+      throw new Error('window object is not valid');
+    }
+
+    if (typeof this.CustomEvent !== 'function') {
+      throw new Error('CustomEvent constructor is not valid');
+    }
+
+    // Binds
+    this.registerListener = this.registerListener.bind(this);
+    this.unregisterListener = this.unregisterListener.bind(this);
+    this.dispatchEvent = this.dispatchEvent.bind(this);
+  }
+
+  /**
+   * The register method that accepts a plugin registry object and returns
+   * the exposedValue of the browser events plugin. The exposedValue is an
+   * object that holds references to the registerListener, unregisterListener,
+   * and dispatchEvent methods.
+   *
+   * @param plugins a plugin registry object.
+   */
+  register(plugins: PluginRegistry): DomEventsPluginExposedValue {
+    this.core = plugins;
+
+    this.exposedValue = {
+      registerListener: this.registerListener,
+      unregisterListener: this.unregisterListener,
+      dispatchEvent: this.dispatchEvent,
+    };
+
+    return this.exposedValue;
+  }
+
+  /**
+   * Register an event listener for a given event and a callback to be
+   * invoked in response to the same event.
+   *
+   * @param eventName Name of the event to be registered/listened for.
+   * @param callback Callback to be registered with the listener.
+   */
+  registerListener(eventName: string, callback: EventListener) {
+    this.window.addEventListener(eventName, callback);
+  }
+
+  /**
+   * Unregister/remove an event listener for the given event as well as
+   * its corresponding callback function.
+   *
+   * @param eventName Name of the event to unregister.
+   * @param callback Callback to be unregistered along with the event.
+   */
+  unregisterListener(eventName: string, callback: EventListener) {
+    this.window.removeEventListener(eventName, callback);
+  }
+
+  /**
+   * Dispatch a given event and provide a payload which will be received
+   * by a listener registered for the given event.
+   *
+   * @param eventName Name of the event to be dispatched.
+   * @param payload Data to accompany the dispatched event.
+   */
+  dispatchEvent(eventName: string, payload?: any) {
+    const eventToDispatch = new this.CustomEvent(eventName, { detail: payload });
+
```

Comment:
I think we can drop this newline. It's only a two-line function.

Reply:
Removed.